# Task 17 - Dataset Exploration

## Iris Dataset

This notebook explores the Iris dataset using Python and Pandas.

In [1]:
import pandas as pd
import seaborn as sns

In [2]:
df = sns.load_dataset("iris")

print("Dataset loaded successfully!")

Dataset loaded successfully!


In [3]:
df

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa
...,...,...,...,...,...
145,6.7,3.0,5.2,2.3,virginica
146,6.3,2.5,5.0,1.9,virginica
147,6.5,3.0,5.2,2.0,virginica
148,6.2,3.4,5.4,2.3,virginica


## 1. Dataset Dimensions

In [4]:
print("Number of rows:", df.shape[0])
print("Number of columns:", df.shape[1])
print("Dataset shape:", df.shape)

Number of rows: 150
Number of columns: 5
Dataset shape: (150, 5)


## 2. Column Information

In [5]:
print("Column names:")
print(df.columns.tolist())

Column names:
['sepal_length', 'sepal_width', 'petal_length', 'petal_width', 'species']


## 3. Dataset Information

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    str    
dtypes: float64(4), str(1)
memory usage: 7.2 KB


## 4. Statistical Summary

In [7]:
df.describe()

,sepal_length,sepal_width,petal_length,petal_width
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


## 5. First Five Observations

In [8]:
df.head()

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


In [9]:
df.head(10)

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa
5,5.4,3.9,1.7,0.4,setosa
6,4.6,3.4,1.4,0.3,setosa
7,5.0,3.4,1.5,0.2,setosa
8,4.4,2.9,1.4,0.2,setosa
9,4.9,3.1,1.5,0.1,setosa


## 6. Last Five Observations

In [10]:
df.tail()

,sepal_length,sepal_width,petal_length,petal_width,species
145,6.7,3.0,5.2,2.3,virginica
146,6.3,2.5,5.0,1.9,virginica
147,6.5,3.0,5.2,2.0,virginica
148,6.2,3.4,5.4,2.3,virginica
149,5.9,3.0,5.1,1.8,virginica


## 7. Missing Values

In [11]:
df.isnull().sum()

sepal_length    0
sepal_width     0
petal_length    0
petal_width     0
species         0
dtype: int64

In [12]:
print("Total missing values:", df.isnull().sum().sum())

Total missing values: 0


## 8. Numerical Columns

In [13]:
numerical_columns = df.select_dtypes(include='number').columns.tolist()

print("Numerical columns:")
print(numerical_columns)

Numerical columns:
['sepal_length', 'sepal_width', 'petal_length', 'petal_width']


## 9. Categorical Columns

In [16]:
# Identify categorical columns
categorical_columns = df.select_dtypes(include=['str']).columns.tolist()

print("Categorical columns:")
print(categorical_columns)

Categorical columns:
['species']


## 10. Unique Species

In [17]:
print(df['species'].unique())

<ArrowStringArray>
['setosa', 'versicolor', 'virginica']
Length: 3, dtype: str


In [18]:
print(df['species'].value_counts())

species
setosa        50
versicolor    50
virginica     50
Name: count, dtype: int64


## 11. Observations

1. The Iris dataset contains 150 rows and 5 columns.

2. The dataset contains four numerical columns:
   sepal_length, sepal_width, petal_length, and petal_width.

3. The species column is a categorical column.

4. The dataset contains no missing values.

5. The dataset contains three different species:
   setosa, versicolor, and virginica.

6. Each species contains 50 observations.

7. The describe() function provides statistical information such as
   mean, standard deviation, minimum, maximum, and quartile values.

## 12. Interview Questions

### Q1. Why should a dataset be explored before modeling?

Dataset exploration is important because it helps us understand the
structure and characteristics of the data before building a machine
learning model. It helps identify data types, missing values,
outliers, distributions, and possible data quality problems.

### Q2. What does DataFrame.shape return?

DataFrame.shape returns a tuple containing the number of rows and
the number of columns in the DataFrame.

For example:

df.shape

returns:

(150, 5)

This means the dataset contains 150 rows and 5 columns.

### Q3. What information does info() provide?

The info() function provides important information about a DataFrame,
including the column names, number of non-null values, data types,
number of entries, and memory usage.

## 13. Conclusion

The Iris dataset was successfully explored using Pandas. 
The dataset contains 150 observations and 5 columns. 
The dataset consists of numerical measurements of the iris flowers 
and a categorical species column. There are no missing values. 
Functions such as shape, columns, info(), describe(), head(), 
and tail() helped us understand the structure and basic 
characteristics of the dataset before modeling.